game-core/src/main/java/games/strategy/triplea/delegate/battle/steps/retreat/EvaderRetreat.java

```diff
@@ -0,0 +1,143 @@
+package games.strategy.triplea.delegate.battle.steps.retreat;
+
+import games.strategy.engine.data.CompositeChange;
+import games.strategy.engine.data.GamePlayer;
+import games.strategy.engine.data.Territory;
+import games.strategy.engine.data.Unit;
+import games.strategy.engine.data.changefactory.ChangeFactory;
+import games.strategy.engine.delegate.IDelegateBridge;
+import games.strategy.triplea.delegate.battle.BattleActions;
+import games.strategy.triplea.delegate.battle.BattleState;
+import games.strategy.triplea.delegate.battle.MustFightBattle;
+import games.strategy.triplea.formatter.MyFormatter;
+import java.util.ArrayList;
+import java.util.Collection;
+import lombok.Builder;
+import lombok.experimental.UtilityClass;
+import org.triplea.sound.SoundUtils;
+
+@UtilityClass
+public class EvaderRetreat {
+
+  @Builder(toBuilder = true)
+  static class Parameters {
+    BattleState battleState;
+    BattleActions battleActions;
+    BattleState.Side side;
+    IDelegateBridge bridge;
+    Collection<Territory> possibleRetreatSites;
+    Collection<Unit> units;
+    String step;
+  }
+
+  public static void retreatUnits(final Parameters parameters) {
+    final BattleState battleState = parameters.battleState;
```

Comment:
Why are all of the parameters being aliased (IE: re-assigned) to new variables up front here? The alternative would be to just use the parameter values inline.

Reply:
I initially didn't have the parameters object but codeclimate complained about too many parameters.  So I switched to the object but didn't want to add `parameters.` in front of all of the places.  Just laziness.  If you want `parameters.` inlined, I'll do it.

Reply:
In the places where it makes sense, perhaps most of them, please do.